unified-runtime/source/adapters/native_cpu/device.cpp

```diff
@@ -193,19 +190,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
   case UR_DEVICE_INFO_IMAGE3D_MAX_DEPTH:
     // Default minimum values required by the SYCL specification.
     return ReturnValue(size_t{2048});
-  case UR_DEVICE_INFO_HALF_FP_CONFIG: {
-    // todo:
-    ur_device_fp_capability_flags_t HalfFPValue = 0;
-    return ReturnValue(HalfFPValue);
-  }
-  case UR_DEVICE_INFO_SINGLE_FP_CONFIG: {
-    // todo
-    ur_device_fp_capability_flags_t SingleFPValue = 0;
-    return ReturnValue(SingleFPValue);
-  }
+  case UR_DEVICE_INFO_HALF_FP_CONFIG:
+  case UR_DEVICE_INFO_SINGLE_FP_CONFIG:
   case UR_DEVICE_INFO_DOUBLE_FP_CONFIG: {
-    ur_device_fp_capability_flags_t DoubleFPValue = 0;
-    return ReturnValue(DoubleFPValue);
+    // All fp types are supported, return minimum flags to indicate support.
+    // TODO: look at this in more detail.
+    ur_device_fp_capability_flags_t SupportedFlags =
+        UR_DEVICE_FP_CAPABILITY_FLAG_DENORM |
```

Comment:
I'm not sure we support denormals on all platforms, we support them if and only if the target supports them. Do we have a way to query that?

Reply:
done https://github.com/intel/llvm/issues/17530